### PR TITLE
Fixes netfoundry/ziti-fabric#30

### PR DESCRIPTION
**Pull request by andrewpmartinez**

### CODE REVIEW

File: controller/controller.go [hunk 1]
```diff
@@ -107,6 +107,12 @@ func (c *Controller) Shutdown() {
 		}
 	}
 
+	if c.config.Db != nil {
+		if err := c.config.Db.Close(); err != nil {
+			pfxlog.Logger().WithError(err).Error("failed to close db")
+		}
+	}
+
 	c.network.Shutdown()
```
Comment (michaelquigley) [inline controller/controller.go]: Wouldn't we want this to run _after_ the `network` has a chance to shut down? It doesn't depend on the database right now, but it probably will soon.

Comment (andrewpmartinez) [reply controller/controller.go]: I can def move it to a `defer`. Didn't consider the shutdown needing to r/w config.

Comment (michaelquigley) [reply controller/controller.go]: You don't necessarily need to use `defer`, couldn't you just move it to happen after `c.network.Shutdown()`?

Comment (andrewpmartinez) [reply controller/controller.go]: I could the only functional difference is if shutdown() panics.

**Pull request merged**